### PR TITLE
Update the-ardupilot-autotest-framework.rst

### PR DESCRIPTION
**Pull request by hendjoshsr71**
Switch all references for vehicles to test.Rover, test.Plane, etc. for clarity when testing different vehicles

### CODE REVIEW

**Comment by peterbarker** [comment]
So one issue here is that the stable releases don't necessarily have those steps, which is why I wasn't chasing this...

**Comment by hendjoshsr71** [comment]
Ahh ok.. I just spent a bit of time figuring how to run specific rover tests as it clearly wasn't `fly.rover` and `drive.Rover` listed here didn't work. It took going into the python script to find the correct keywords. 

Is there a better way to document stable vs master tests? 

As `build.'vehicle_name'` and `test.'vehicle_name'` are far easier than to use without looking up `dive.Sub` or `drive.APMrover2`, etc.

**Comment by khancyr** [comment]
use Tab completion , it should autocomplete for you :-P

**Comment by peterbarker** [comment]
On Mon, 11 Jan 2021, Josh Henderson wrote:

> Is there a better way to document stable vs master tests?

Wait for new stable releases to be made?  :-)

Could also change all the invocations in there but add ".. note:"s to say 
that older software may use build.APMrover2 drive.APMrover2 etc

> As build.'vehicle_name' and test.'vehicle_name' are far easier than to use without looking up dive.Sub or drive.APMrover2, etc.

I thought so.

I *think* both stable and unstable autotest.py s have various options to 
list the available steps.

Try --help (again, don't run autotest.py with no options ;-) )

It's probably a bit late to be back-porting things to make stable work 
like master....

**Comment by hendjoshsr71** [comment]
Updated it a bit.. just trying to help newbs like myself out :)

I did read --help 

I saw the `--list` that lists test.Copter , etc. unfortunately for me I started at the bottom and tried  `--list-vehicles` and `--list-vehicles-test` first such is life and learning!

I might add the change to `--help` to make it clearer what "steps" means as naively I ignored it.  
`--list                list the available steps: build, test, run, defaults, etc`

**Review by peterbarker — CHANGES_REQUESTED** [review]
LGTM apart from the "they still work bit" :-)

File: dev/source/docs/the-ardupilot-autotest-framework.rst [hunk 1]
```diff
@@ -81,16 +81,21 @@ Help is available:
...
 
 .. toctree::
     :maxdepth: 1
 
      AutoTest Sample Output <autotest-verbose>
-     
-A list of available steps is available with ``--list``.
+
+.. note:: 
+   Older versions of ``autotest.py`` used these steps (still available): ``build.ArduPlane build.ArduCopter build.APMrover2 build.ArduSub build.AntennaTracker fly.ArduCopter
```
[the remark this answers is not in the document]
Comment (hendjoshsr71) [reply dev/source/docs/the-ardupilot-autotest-framework.rst]: LOL!

**Review by peterbarker — APPROVED** [review]
LGTM, thanks

**Comment by Hwurzburg** [comment]
thanks...merging

**Pull request merged**